cli/deployment/config.go

```diff
@@ -374,6 +374,19 @@ func newConfig() *codersdk.DeploymentConfig {
 			Usage: "Controls if the 'Secure' property is set on browser session cookies.",
 			Flag:  "secure-auth-cookie",
 		},
+		StrictTransportSecurity: &codersdk.DeploymentConfigField[int]{
+			Name: "Strict-Transport-Security",
+			Usage: "Controls if the 'Strict-Transport-Security' header is set on all static file responses. " +
+				"This header should only be set if the server is accessed via HTTPS. The value should be a whole number in seconds.",
```

Comment:
I think it would be good to explain that the number sets the maxAge here.

> The time, in seconds, that the browser should remember that a site is only to be accessed using HTTPS.

We could also consider splitting this up into an enable bool and explicit maxage option that's required when enabled.

Reply:
I added that in the message. If I add another flag, then it's 3 flags to turn this thing fully on :(. I figure anything >0 make sense to enable it.